docs/codeql/codeql-cli/extractor-options.rst

```diff
@@ -0,0 +1,136 @@
+.. extractor-options:
+
+Extractor options
+=================
+
+The CodeQL CLI uses special programs, called extractors, to extract information from the source code of a
+software system into a database that can be queried. You can customize the behavior of extractors by
+setting extractor configuration options through the CodeQL CLI.
+
+About extractor options
+-----------------------
+
+Each extractor defines its own set of configuration options. To find out which options are available for a particular extractor, you can run ``codeql resolve languages`` or ``codeql resolve extractor`` with the ``--format=betterjson`` option. The ``betterjson`` output format provides the root paths of extractors and additional information. The output of ``codeql resolve extractor --format=betterjson`` will often be formatted like the following example::
+
+    {
+         "extractor_root" : "/home/user/codeql/java",
+         "extractor_options" : {
+             "option1" : {
+             "title" : "Java extractor option 1",
+             "description" : "An example string option for the Java extractor.",
+             "type" : "string",
+             "pattern" : "[a-z]+"
+             },
+         "group1" : {
+             "title" : "Java extractor group 1",
+             "description" : "An example option group for the Java extractor.",
+             "type" : "object",
+             "properties" : {
+                 "option2" : {
+                     "title" : "Java extractor option 2",
+                     "description" : "An example array option for the Java extractor",
+                     "type" : "array",
+                     "pattern" : "[1-9][0-9]*"
+                 }
+             }
+         }
+         }
+     }
+
+The extractor option names and descriptions are listed under ``extractor_options``. Each description may contain the following fields::
```

Comment:
This might be minor and not necessary, but it could be good to use a different word than 'description' since that is also a field name, and instead use the word 'option' or something else?
```suggestion
The extractor option names and descriptions are listed under ``extractor_options``. Each option may contain the following fields::
```